### PR TITLE
Modernize API

### PR DESCRIPTION
**Pull request by philium**
Uses async/await instead of completion handlers. Also eliminates GCD queues.

### CODE REVIEW

File: Sources/Dejavu/GRDBSession.swift [hunk 1]
```diff
@@ -630,3 +493,192 @@ final class GRDBSession: SessionInternal, @unchecked Sendable {
...
+        var request: URLRequest
+        var dejavuRequest: Request
+        var instanceCount: Int
+        var response: URLResponse?
+        var data: Data?
+        var error: Error?
+    }
+}
+
+extension GRDBSession: DejavuNetworkObservationHandler {
```
Comment (rolson) [inline Sources/Dejavu/GRDBSession.swift:508]: The GRDBSession for observation handler is here, but the extension for interception handler is in another file. Should both be in here? It doesn't seem like that other file needs to know about GRDB.

Comment (philium) [reply Sources/Dejavu/GRDBSession.swift:508]: Yes, they could both be in here. I actually preferred to keep them both out (leaving this file smaller), but this one needed to be in this file in order to efficiently update the private state.

File: Sources/Dejavu/GRDBSession.swift [hunk 2]
```diff
@@ -14,78 +14,87 @@
...
-                // otherwise open it
-                dbQueue = try? DatabaseQueue(path: configuration.fileURL.path)
+                dbQueue = try Self.makeDatabaseAndSchema(fileURL: temporaryURL)
             }
         }
+        
+        self.configuration = configuration
+    }
+    
+    func clearCache() {
```
Comment (pgruenler) [inline Sources/Dejavu/GRDBSession.swift:81]: I'm not seeing this function get called when I'm setting the mode to `cleanRecord` and new databases are not getting created.

Comment (philium) [reply Sources/Dejavu/GRDBSession.swift:81]: This function is not being called, either internally or externally, which is why we're deprecated it. One of the updates in this PR is to perform database operations in a temporary directory and then remove/replace the old database when the session ends. So for `cleanRecord`, that is now done in `end()`. However, [one of the URLs](https://github.com/Esri/Dejavu/blob/7301e9214d7c51f4b93b894bf5ba6610f7ce7767/Sources/Dejavu/GRDBSession.swift#L318) being used isn't correct, so I'll update the PR to fix that. Thanks for finding that!

Comment (rolson) [reply Sources/Dejavu/GRDBSession.swift:81]: I tested cleanRecord and it worked for me. Is it still not working for you?

**Review by rolson — APPROVED** [review]

**Pull request merged**